packages/api/src/controllers/releaseBadgeController.js

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import ReleaseBadgeModel from '../models/releaseBadgeModel.js';
+
+const releaseBadgeController = {
+  getReleaseBadgeVersion() {
+    return async (req, res) => {
+      try {
+        const { user_id } = req.auth;
+        const record = await ReleaseBadgeModel.query().findById(user_id);
+
+        if (record) {
+          res.json({ app_version: record.app_version });
+        } else {
+          res.json({ app_version: null });
+        }
+      } catch (error) {
+        console.error(error);
+        res.status(400).json({ error });
+      }
+    };
+  },
+  updateReleaseBadgeVersion() {
+    return async (req, res) => {
+      try {
+        const { user_id } = req.auth;
+        const { app_version } = req.body;
+        const existingVersion = await ReleaseBadgeModel.query().findById(user_id);
+
+        if (existingVersion) {
+          await ReleaseBadgeModel.query()
+            .findById(user_id)
+            .patch({ app_version })
+            .context({ user_id });
+        } else {
+          await ReleaseBadgeModel.query()
+            .insert({
+              user_id,
+              app_version,
+            })
+            .context({ user_id });
+        }
+
+        res.sendStatus(200);
+      } catch (error) {
+        console.error(error);
+        res.status(400).json({ error });
```

Comment:
I think we do this a lot throughout the app, but since we don't have a reason to think here that the error is caused by something that was sent wrong by the client, I don't think "bad request" is the appropriate status code. For a catch all for generic errors like this, I think 500 makes more sense. Wdyt?

Reply:
Ah this is something I'd love to establish guidelines for! I think we have always used 400 as our catch-all, with just a few 500s sprinkled in randomly. I don't know if it's intended, but as a side-effect of our habit now I associate 500 errors as being "severe" ones, since they are usually only sent by the general error handler in `server.js`. That said I think I threw one into a controller recently (finance report) so I'm not even internally consistent 😔

Reply:
The code shouldn't really be related to the severity, but to the cause of the error. 400 codes indicate something went wrong on the client side, so for example if you check for a required parameter and it's missing in the request, or the shape of the payload doesn't look the way it should be, or there's a missing header -- anything that indicates that the client sent out something you weren't expecting. Reading the definition for the 500 status code (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500), it says this is the "catch all" for a server error where you don't really know what went wrong, so I think that would be the fitting one in these generic catch blocks. Not a blocker for merging though! But something I think we should start doing throughout the app